pkg/eks/tasks.go

```diff
@@ -356,7 +356,7 @@ func (c *ClusterProvider) ClusterTasksForNodeGroups(cfg *api.ClusterConfig, inst
 			logger.Info("\t see the following page for instructions: https://awsdocs-neuron.readthedocs-hosted.com/en/latest/neuron-deploy/tutorials/tutorial-k8s.html#tutorial-k8s-env-setup-for-neuron")
 		}
 	}
-	if haveNvidiaInstanceType {
+	if haveNvidiaInstanceType && clusterRequiresDevicePlugin(cfg) {
```

Comment:
Doesn't this mean if my cluster is made up of:
- A regular non-GPU AL2 node
- GPU NVIDIA Bottlerocket node

that it will install the device plugin? Is this the desired behaviour? I just wanna check as I'm not 100% sure I understand when we do/don't need the plugin installed. I would of thought given a non-gpu AL2 only cluster doesn't need it, and a GPU only bottlerocket cluster also doesn't need it, that the combination of the two in the same cluster also wouldn't need it :thinking: 

I validated that it did get installed:
```
cat cluster.yaml
---
apiVersion: eksctl.io/v1alpha5
kind: ClusterConfig

metadata:
  name: br-gpu-and-al2-nor
  region: us-west-2
  version: '1.21'

nodeGroups:
  - name: br-nvidia
    instanceType: p3.2xlarge
    desiredCapacity: 1
    amiFamily: Bottlerocket
    bottlerocket:
      settings:
        motd: "Hello from eksctl!"
        host-containers:
          admin:
            enabled: true
  - name: al2-normal
    instanceType: m5.large
    desiredCapacity: 1
```

```
k get daemonsets.apps -A
NAMESPACE     NAME                             DESIRED   CURRENT   READY   UP-TO-DATE   AVAILABLE   NODE SELECTOR   AGE
kube-system   aws-node                         2         2         2       2            2           <none>          50m
kube-system   kube-proxy                       2         2         2       2            2           <none>          50m
kube-system   nvidia-device-plugin-daemonset   2         2         2       2            2           <none>          39m
```

Reply:
@aclevername , correct, what you described is what should happen, according to @cPu1 the device plugin should only be installed in AL2 instances, I'll update the code to reflect this. I'd assume that the Neuron device plugin should only be installed in AL2 instances, I'll make the changes to reflect this as well, since I could do a little refactor to remove the function that I added now that I know the conditions required to install the device plugins.

Reply:
> @aclevername , correct, what you described is what should happen, according to @cPu1 the device plugin should only be installed in AL2 instances

I think @cPu1 meant only GPU AL2 Instances? From what I'm reading here https://github.com/weaveworks/eksctl/issues/4760#issuecomment-1040308851

Reply:
Ok, I'll add a different condition somewhere (if it doesn't exist already), so that folks don't try to install Bottlerocket on inferentia hosts because we don't support it just yet. It is a little weird to me that only AL2 hosts require the NVIDIA GPU device plugin, but it is OK to install the Neuron device plugin on all other hosts, that's why I thought the Neuron device plugin would follow the same rule as the NVIDIA GPU device plugin.

Reply:
I might be wrong, I can't find clear documentation on which nodes actually need to have the plugin running, wether it's just the GPU nodes themselves or all nodes. If it's all nodes then the current code you have is correct. If it's only GPU nodes then we only want to install the nvidia plugin if the cluster contains a AL2 GPU node. 

Does that make senses? I'm sorry I don't have more information :/

Reply:
Ideally GPU nodes should have the plugin running only if the AMI family supports them, because the device plugin relies on the driver provided by the AMI. Right, now I don't know for sure if AMIs other than AL2 or Bottlerocket support NVIDIA GPUs or Inferentia hardware. The [unit tests](https://github.com/weaveworks/eksctl/blob/57d0afb297742a579391fd3a8fa6ccafcaad3f10/pkg/apis/eksctl.io/v1alpha5/gpu_validation_test.go#L58) suggest that only AL2 supports both NVIDIA and Inferientia instances (and with my change, Bottlerocket will support NVIDIA GPUs).

I think for now my change should only cover installing the device plugin when there is a nodegroup with AL2 with NVIDIA instances, and then we could revisit what other combinations of AMI families/instance types require device plugins.